operator/pkg/operator.go

```diff
@@ -497,15 +499,33 @@ func (o *Operator) verify(verificationData VerificationData, results chan bool)
 
 	case common.SP1:
 		verificationResult := sp1.VerifySp1Proof(verificationData.Proof, verificationData.VmProgramCode)
-		o.Logger.Infof("SP1 proof verification result: %t", verificationResult)
-		results <- verificationResult
+		if !verificationResult {
+			o.Logger.Infof("SP1 proof verification failed. Trying old SP1 version...")
+			verificationResult = sp1_old.VerifySp1ProofOld(verificationData.Proof, verificationData.VmProgramCode)
+			if !verificationResult {
+				o.Logger.Errorf("SP1 proof verification failed")
+				results <- verificationResult
+			}
+		} else {
+			o.Logger.Infof("SP1 proof verification result: %t", verificationResult)
+			results <- verificationResult
+		}
```

Comment:
There is an unsent verificationResult case

```suggestion
		if !verificationResult {
			o.Logger.Infof("SP1 proof verification failed. Trying old SP1 version...")
			verificationResult = sp1_old.VerifySp1ProofOld(verificationData.Proof, verificationData.VmProgramCode)
			if !verificationResult {
				o.Logger.Errorf("SP1_old proof verification failed")
			}
		}
		o.Logger.Infof("SP1 proof verification result: %t", verificationResult)
		results <- verificationResult
```